src/main/java/de/tum/in/www1/artemis/service/UserService.java

```diff
@@ -325,6 +311,35 @@ private boolean removeNonActivatedUser(User existingUser) {
         return true;
     }
 
+    /**
+     * searches the (optional) LDAP service for a user with the give registration number (= Matrikelnummer) and returns a new Artemis user
+     * Note: this method should only be used if the user does not yet exist in the database
+     *
+     * @param registrationNumber the matriculation number of the student
+     * @return a new user or null if the LDAP user was not found
+     */
+    public Optional<User> createUserFromLdap(String registrationNumber) {
+        if (ldapUserService.isPresent()) {
+            Optional<LdapUserDto> ldapUserOptional = ldapUserService.get().findByRegistrationNumber(registrationNumber);
+            if (ldapUserOptional.isPresent()) {
+                LdapUserDto ldapUser = ldapUserOptional.get();
+                log.info("Ldap User " + ldapUser.getUsername() + " has registration number: " + ldapUser.getRegistrationNumber());
+                // Use empty password, so that we don't store the credentials of Jira users in the Artemis DB
+                User user = createUser(ldapUser.getUsername(), "", ldapUser.getFirstName(), ldapUser.getLastName(), ldapUser.getEmail(), null, "en");
```

Comment:
Maybe it would be a good idea to create the users with a randomised password and send it to them via email. As the users will access their exams from Artemis, there might be data privacy concerns if their accounts are left with an empty password. To my knowledge, it is not possible to change the password later in Artemis? 
This extension could be part of a follow-up PR.

Reply:
This is intentional. Students use their TUM LDAP password, which is **NOT** stored in the Artemis database (therefore we have an empty password here). In the TUM setup it works as follows: during login, Artemis asks JIRA, which in turn asks the TUM LDAP. If the password is correct, the user is logged in, otherwise an error message is shown. So basically, for TUM users, the password column in the database is ignore. We can, however, not set it null, because then Spring would complain